From ae3b2bc1fdcac54e51bc91e9ea7f25b9a022bc2c Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Wed, 29 Aug 2012 13:25:47 +0000
Subject: [PATCH 0338/1077] mcorba.lib, mcorbas.lib V1.0.6.19 bugfix, with
 ORBClientConnectionHandler RW, ACE TAO ignores CloseConnection-Messages   use
 MT_NOUPCALL instead

git-svn-id: svn://localhost/SelfServiceCommon/trunk@404 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/cpp/MassaiCorba/inc/massaiCORBA_def.h     |  2 ++
 .../Massai/cpp/MassaiCorba/src/orbimpl_tao.cpp       | 12 +-----------
 2 files changed, 3 insertions(+), 11 deletions(-)

diff --git a/SelfServiceCommon/Massai/cpp/MassaiCorba/inc/massaiCORBA_def.h b/SelfServiceCommon/Massai/cpp/MassaiCorba/inc/massaiCORBA_def.h
index f3b1fe9d..e931ce01 100644
--- a/SelfServiceCommon/Massai/cpp/MassaiCorba/inc/massaiCORBA_def.h
+++ b/SelfServiceCommon/Massai/cpp/MassaiCorba/inc/massaiCORBA_def.h
@@ -22,6 +22,8 @@
  *                   \li SteT - bugfix, Mantis 24158 call ACE::init, ACE::fini around use of ORB_init, ORB::destroy
  *                              bugfix, use correct POA for servant deactivation
  *                              bugfix, fixed reading port after hostname in configuration from CORBA.SERVANTS.*
+ *                              bugfix, with ORBClientConnectionHandler RW, ACE TAO ignores CloseConnection-Messages
+ *                                      use MT_NOUPCALL instead
  *
  *  \date 13.03.2012 \li V1.0.6.18
  *                   \li SteT - enhancement, Mantis 0018942: mCorba: creation order of servant orb and client orb and synchronisation problems
diff --git a/SelfServiceCommon/Massai/cpp/MassaiCorba/src/orbimpl_tao.cpp b/SelfServiceCommon/Massai/cpp/MassaiCorba/src/orbimpl_tao.cpp
index 29402fb9..5c6c0713 100644
--- a/SelfServiceCommon/Massai/cpp/MassaiCorba/src/orbimpl_tao.cpp
+++ b/SelfServiceCommon/Massai/cpp/MassaiCorba/src/orbimpl_tao.cpp
@@ -12,7 +12,6 @@
 #define USE_INVOCATION_TIMEOUT
 #endif
 #if defined(_TAO_1_5_) || defined (_TAO_2_0_)
-#define USE_CONNECTION_HANDLER_CLEANUP
 #define USE_KEEPALIVE
 #endif
 
@@ -43,16 +42,7 @@ namespace
   void preventNestedCalls(ParameterList& parameterList)
   {
     parameterList.push_back("-ORBSvcConfDirective");
-    parameterList.push_back("static Client_Strategy_Factory \""
-      "-ORBClientConnectionHandler rw "
-      "-ORBTransportMuxStrategy exclusive "
-      "-ORBConnectStrategy blocked "
-#if defined(USE_CONNECTION_HANDLER_CLEANUP)
-      "-ORBConnectionHandlerCleanup 1 "
-#endif
-      "\"");
-    parameterList.push_back("-ORBSvcConfDirective");
-    parameterList.push_back("static Resource_Factory \"-ORBFlushingStrategy blocking\"");
+    parameterList.push_back("static Client_Strategy_Factory \"-ORBClientConnectionHandler MT_NOUPCALL\"");
   }
 
   void setORBEndpoint(ParameterList& parameterList, const std::string& endpoint)
-- 
2.41.0.windows.1

